StandardLibraryEssentials/StandardTemplateLibrary/STLAlgorithms/task.md

```diff
@@ -0,0 +1,179 @@
+```cpp
+std::fill(start_iterator, end_iterator, value);
+```
+The first two arguments are iterators that define the range of elements to be filled. The third argument is the value to be filled.
+```cpp
+#include <iostream>
+#include <vector>
+#include <algorithm>
+
+int main() {
+    std::vector<int> v = {1, 2, 3, 4, 5};
+    std::fill(v.begin(), v.end(), 0);
+    // v = {0, 0, 0, 0, 0}
+}
+```
+
+Also, there is a header called [`<numeric>`](https://en.cppreference.com/w/cpp/header/numeric) that contains a number of useful algorithms for numeric operations.
+
+#### 1. `std::accumulate()`
+[`std::accumulate()`](https://en.cppreference.com/w/cpp/algorithm/accumulate) is used to calculate the sum of a range of elements. It is implemented as a simple linear `O(n)` traversal, with this syntax:
+```cpp
+std::accumulate(start_iterator, end_iterator, initial_value);
+```
+The first two arguments are iterators that define the range to be summed. The third argument is the initial value of the sum. Be aware of the accumulator type, as it may overflow if the sum is too large. For example, if you want to sum a range of integers, you can use `0LL` as the initial value to avoid overflow.
+```cpp
+#include <iostream>
+#include <vector>
+#include <numeric>
+
+int main() {
+    std::vector<int> v = {1, 2, 3, 4, 5, 6, 7, 8, 9};
+    long long sum = std::accumulate(v.begin(), v.end(), 0LL);
+    std::cout << "Sum: " << sum << std::endl;
+}
+```
+
+#### 2. `std::gcd()` and `std::lcm()`
+[`std::gcd()`](https://en.cppreference.com/w/cpp/numeric/gcd) and [`std::lcm()`](https://en.cppreference.com/w/cpp/numeric/lcm) are used to calculate the Greatest Common Divisor and the Least Common Multiple of two integers. They are implemented as `O(log n)` algorithms, looking like that:
+```cpp
+std::gcd(a, b);
+std::lcm(a, b);
+```
+The two arguments are the two integers, with which the GCD or LCM is calculated.
+```cpp
+#include <iostream>
+#include <numeric>
+
+int main() {
+    int a = 12, b = 18;
+    std::cout << "GCD: " << std::gcd(a, b) << std::endl;
+    std::cout << "LCM: " << std::lcm(a, b) << std::endl;
+}
+```
+
+Finally, let's get some practice with these algorithms. Your task is to implement the following functions:
+ - `create_and_fill` - create a vector of size `n` and fill it with the values from `1` to `n` in ascending order.
+ - `shuffle_vector` - shuffle the elements of the vector.
+ - `sort_descending` - sort the elements of the vector in descending order.
+ - `reverse_vector` - reverse the elements of the vector.
+ - `find_value` - find the first occurrence of the value `x` in the vector.
+ - `find_sum` - calculate the sum of the elements in the vector.
+ - `find_gcd` - calculate the GCD of sum of the elements in the given vector and the value `x`.
```

Comment:
Instead of this small tasks, let's think of a single, more complex task that would require to utilize several algorithms at once. 

For example, a function, that would calculate a sum of elements in the collection with values between two provided values `lowerBound` and `upperBound`. 

To implement this, the student will have to first sort the collection, then get two iterators using `find`, and finally use `accumulate` to calculate the sum.